src/wallet/rpcwallet.cpp

```diff
@@ -4063,6 +4071,14 @@ UniValue z_sendmany(const UniValue& params, bool fHelp)
                         RPC_INVALID_PARAMETER,
                         "Cannot send between Sprout and Sapling addresses using z_sendmany");
                 }
+
+                int nextBlockHeight = chainActive.Height() + 1;
+
+                const bool canopyActive = Params().GetConsensus().NetworkUpgradeActive(nextBlockHeight, Consensus::UPGRADE_CANOPY);
+
+                if ((fromTaddr && toSprout) && canopyActive) {
```

Comment:
Perhaps move the `const bool canopyActive = ...` inside the `if` so it's not computed unless needed? That is, something like:
```
if (fromTaddr && toSprout) {
    const bool canopyActive = ...
    if (canopyActive) {
        throw ...
    }
}